src/main/kotlin/telegram/Menu.kt

```diff
@@ -195,124 +284,375 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
+    )
+}
+
+internal fun <T: Any> showPaginatedMenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    title: String,
+    list: List<T>,
+    actionForEach: KeyboardContext.(T) -> Unit,
+    bottomContent: KeyboardContext.(Long) -> Unit,
+    menuCommand: Command,
+    desiredPageIndex: Int,
+    itemsPerPage: Int = defaultItemsPerPage
+) {
+    val msgId = if (messageId == -1L) {
+        sendMessage(
+            bot,
+            chatId,
+            title
+        )
+    } else {
+        messageId
+    }
+    updateMessage(
+        bot,
+        chatId,
+        msgId,
         replyMarkup = inlineKeyboard {
-            button(blankCommand named "Список администраторов")
-            admins.find().forEach {
-                accounts.get(it.chatId)?.let { acc ->
+            button(blankCommand named title)
+            val listSize = list.size
+            if (listSize > 0) {
+                val quotient = listSize / itemsPerPage
+                val totalAvailablePages = if (listSize % itemsPerPage == 0) {
+                    quotient
+                } else {
+                    quotient + 1
+                }
+                val pageIndex = if (desiredPageIndex >= totalAvailablePages || desiredPageIndex < 0) {
+                    0
+                } else {
+                    desiredPageIndex
+                }
+                button(blankCommand named "Номер страницы: ${pageIndex + 1}")
+                val firstElementIndex = pageIndex * itemsPerPage
+                row {
+                    if (pageIndex > 0) {
+                        button(menuCommand named "⬅", msgId, pageIndex - 1)
+                    }
+                    if (pageIndex < totalAvailablePages - 1) {
+                        button(menuCommand named "➡", msgId, pageIndex + 1)
+                    }
+                }
+                for (i in firstElementIndex until firstElementIndex + itemsPerPage) {
+                    if (i < listSize) {
+                        actionForEach(list.get(i))
+                    }
+                }
+                if (totalAvailablePages > 1) {
                     row {
-                        button(blankCommand named acc.fullName())
-                        button(removeAdminCommand, it.chatId, messageId)
+                        button(menuCommand named "Первая", msgId, 0)
+                        button(menuCommand named "Последняя", msgId, totalAvailablePages - 1)
                     }
                 }
+            } else {
+                button(blankCommand named "🤷 Пусто")
             }
-            button(adminBackCommand, messageId)
+            bottomContent(msgId)
         }
     )
 }
 
-internal fun showGameStatusMenu(chatId: Long, messageId: Long, bot: Bot) {
-    bot.editMessageReplyMarkup(
-        ChatId.fromId(chatId),
+internal fun <K: Any, T: Any> showPaginatedMenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    title: String,
+    list: Collection<K, T>,
```

Comment:
Combination of UI logic and db access in one function is a bad practice, it goes against single responsibility principal